swift/apple/FirezoneKit/Sources/FirezoneKit/Models/Configuration.swift

```diff
@@ -1,99 +1,178 @@
+//
+//  Configuration.swift
+//  (c) 2024 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+//  A thin wrapper around UserDefaults for user and admin managed app configuration.
+
 import Foundation
 
-public class Configuration: Codable {
+#if os(macOS)
+import ServiceManagement
+#endif
+
+@MainActor
+public class Configuration: ObservableObject {
+  static let shared = Configuration()
+
+  @Published private(set) var publishedInternetResourceEnabled = false
```

Comment:
We need separate Published properties here because:

- computed properties cannot be published
- the observers need the _actual_ changed value - observing `configuration.objectWillChange` doesn't provide that